File: etc/test/install_init_test.pl
#!/usr/bin/perl

use strict;
use warnings;
use FindBin;
use Cwd;
use Test::More;

my $root = Cwd::abs_path($FindBin::Bin . "/../");
like "$root", qr/dotfiles\/(?:(?!\/).)*$/, 'within dotfiles/etc';

my @list = glob "$root/init/{,osx/}*.sh";
cmp_ok @list, '>', 0, '@list > 0';

foreach my $f (@list) {
    # Check debug mode init scripts
    system("DEBUG=1 bash $f >/dev/null");

    # Shorten path
    #$f =~ s/^.*dotfiles\///;
    #ok $? == 0, sub{my $a = shift; $a =~ s/^.*dotfiles\///; $a}->($f);
    #ok $? == 0, sub{s/^.*dotfiles\///; $_}->($f);
    ok $? == 0, sub($) {
        my $v = shift;
        $v =~ s/^.*dotfiles\///; $v;
    }->($f);
}
done_testing;

File: etc/test/disuse/install_init_test.rb
#!/usr/bin/env ruby

def test_initialize
    err = []
    dirs = Dir.glob("etc/init/*.sh") + Dir.glob("etc/init/osx/*.sh")
    dirs.each {|d|
        unless system("DEBUG=1 bash #{d} >/dev/null") && $? == 0 then
            err.push([d])
        end
    }
    exit 1 unless err.length == 0
end

test_initialize
